### PR TITLE
Add rule: truthy, to forbid truthy values that are not quoted.

### PR DESCRIPTION
**Pull request by pdericson**

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def test_run_non_ascii_file(self):
         # Make sure the default localization conditions on this "system"
         # support UTF-8 encoding.
         loc = locale.getlocale()
-        locale.setlocale(locale.LC_ALL, 'C.UTF-8')
+        locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
```
Comment (pdericson) [inline tests/test_cli.py]: `C.UTF-8` didn't work for me on OS X...

Comment (adrienverge) [inline tests/test_cli.py]: Please don't include this is this pull-request, we can do another one if needed.

(By the way `en_US.UTF-8` isn't supposed to be present on non-English-speaking systems, and I know it isn't on Debian build system.)

File: yamllint/conf/relaxed.yaml [hunk 2]
```diff
@@ -26,3 +26,5 @@ rules:
   line-length:
     level: warning
     allow-non-breakable-inline-mappings: yes
+  truthy:
+    level: warning
```
Comment (adrienverge) [inline yamllint/conf/relaxed.yaml]: I'd rather have it disabled in `relaxed` rule. I know a bunch of people that are really attached to their `yes`/`no` in Ansible...

File: tests/rules/test_truthy.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 Adrien Vergé
```
Comment (adrienverge) [inline tests/rules/test_truthy.py]: Hey! I didn't write this. ;)

File: yamllint/rules/truthy.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 Adrien Vergé
```
Comment (adrienverge) [inline yamllint/rules/truthy.py]: Please put your name here :)

File: tests/test_spec_examples.py [hunk 5]
```diff
@@ -119,6 +120,9 @@ class SpecificationTestCase(RuleTestCase):
     'example-8.22': ('indentation: disable\n'),
     'example-10.1': ('colons: {max-spaces-before: 2}\n'),
     'example-10.2': ('indentation: {indent-sequences: no}\n'),
+    'example-10.5': ('truthy: disable\n'),
```
Comment (adrienverge) [inline tests/test_spec_examples.py]: Why this one?

File: tests/test_spec_examples.py [hunk 6]
```diff
@@ -65,6 +65,7 @@ class SpecificationTestCase(RuleTestCase):
     'example-2.16': ('empty-lines: {max-end: 1}\n'),
     'example-2.18': ('empty-lines: {max-end: 1}\n'),
     'example-2.19': ('empty-lines: {max-end: 1}\n'),
+    'example-2.21': ('truthy: disable\n'),
```
Comment (adrienverge) [inline tests/test_spec_examples.py]: Why this one?

File: yamllint/rules/truthy.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+          'FALSE', 'False',  # false is a boolean
+          'ON', 'On', 'on',
+          'OFF', 'Off', 'off']
+
+
+def check(conf, token, prev, next, nextnext, context):
+    if isinstance(token, yaml.tokens.ScalarToken):
+        if token.value in TRUTHY and token.style is None:
+            yield LintProblem(token.start_mark.line + 1,
+                              max(1, token.start_mark.column),
```
Comment (adrienverge) [inline yamllint/rules/truthy.py]: This should be `token.start_mark.column + 1` (and you can remove `max()`).

File: tests/rules/test_truthy.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        conf = 'truthy: disable'
+        self.check('---\n1: True\n', conf)
+        self.check('---\nTrue: 1\n', conf)
+
+    def test_enabled(self):
+        conf = 'truthy: enable\n'
+        self.check('---\n1: True\n', conf, problem=(2, 3))
+        self.check('---\nTrue: 1\n', conf, problem=(2, 1))
+        self.check('---\n1: "True"\n', conf)
+        self.check('---\n"True": 1\n', conf)
```
Comment (adrienverge) [inline tests/rules/test_truthy.py]: Could you add this?
```py
        self.check('---\n'
                   '!!seq [\n'
                   '  !!bool true, !!bool false,\n'
                   '  !!bool "false", !!bool "FALSE",\n'
                   '  !!bool "true", !!bool "True",\n'
                   '  !!bool "false", !!bool "FALSE",\n'
                   ']\n', conf)
        self.check('---\n'
                   '!!seq [\n'
                   '  !!bool true, !!bool True,\n'
                   ']\n', conf, problem=(3, 23))
```

File: tests/test_cli.py (hunk 1 above)
Comment (pdericson) [inline tests/test_cli.py]: I have reverted this change, I can run the tests in docker so all good

```
docker run -v $PWD:/mnt -w /mnt python:2.7 python setup.py test
```

File: yamllint/rules/truthy.py (hunk 7 above)
Comment (pdericson) [inline yamllint/rules/truthy.py]: I copied this from commas.py, I can see why it's invalid now :)

**Comment by adrienverge** [comment]
That's better, thanks!

Have you got a chance to consider my other remarks? (the extra tests with `!!bool` and the spec examples you disable -- I think two of them are working fine and shouldn't be disabled)

File: tests/rules/test_truthy.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        self.check('---\n'
+                   '!!seq [\n'
+                   '  !!bool true, !!bool false,\n'
+                   '  !!bool "false", !!bool "FALSE",\n'
+                   '  !!bool "true", !!bool "True",\n'
+                   '  !!bool "false", !!bool "FALSE",\n'
+                   ']\n', conf)
+        self.check('---\n'
+                   '!!seq [\n'
+                   '  !!bool true, !!bool True,\n'
```
Comment (pdericson) [inline tests/rules/test_truthy.py]: @adrienverge I'm wondering if `!!bool True` should be allowed by the truthy rule.  I haven't figured out yet if the explicit tag type is available to inspect or is lost during parsing by pyyaml...

**Comment by pdericson** [comment]
Sorry, my session ended abruptly yesterday, I was addressing feedback during my commute

**Comment by adrienverge** [comment]
@pdericson this is great, thanks for your contribution!

**Comment by ghiculescu** [comment]
just a quick heads up that the version of `yamllint` that apt-get installs doesnt include this rule. the pip version does.

**Comment by adrienverge** [comment]
@ghiculescu Are you using Debian? Ubuntu?

@debalance is kindly maintaining packages for Debian, which means new versions should also end up in next Ubuntu versions. For really up-to-date versions, we would need an Ubuntu maintainer (help wanted!)

**Pull request merged**